src/neo-vm/ExecutionEngine.cs

```diff
@@ -277,17 +277,20 @@ private bool ExecuteInstruction()
                         if (!ExecuteCall(x.Position)) return false;
                         break;
                     }
-                case OpCode.THROW:
-                    {
-                        return false;
-                    }
-                case OpCode.THROWIF:
-                case OpCode.THROWIFNOT:
+                case OpCode.ASSERT:
                     {
                         if (!TryPop(out bool x)) return false;
-                        if (x ^ (instruction.OpCode == OpCode.THROWIFNOT)) return false;
+                        if (!x)
+                        {
+                            State = VMState.FAULT;
+                            return false;
```

Comment:
If you return `false`, the state will be automatically changed to `FAULT`. So you don't need to set it explicitly.